### ISSUE TITLE
Try replacing Nokogiri with Oga, Ox, or similar

### ISSUE THREAD

**Issue by bryanp** [pakyow/pakyow#124]
I'd like to see if we can use [Oga](https://github.com/YorickPeterse/oga) instead of Nokogiri. Doing this will remove the libxml dependency, which should simplify the installation of Pakyow.

The only thing Pakyow uses Nokogiri for is parsing an HTML document and iterating over nodes when creating a `StringDoc`. I can think of two considerations here:

1. Does Oga work? I've not had a lot of experience with it.
2. What's the performance implication?

Consider this an experiment. I'd like to try the change and see what we learn. If it turns out to be a good step forward we can include it in the next release.

**Comment by iamvery** [comment]
Fwiw, I recently replaced Nokogiri with https://github.com/ohler55/ox (another pure-Ruby XML parser). In our particular case, we were parsing some thousands of hundreds MBs XML documents. We saw a non-negligible speed increase just by swapping the library. I can't recall the specifics, but I seems to think it was somewhere between 20% and 60% speed increase over Nokogiri.

Not saying it should be used over Oga, just offering another option in case it wasn't known :)
